app/services/provider_services.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Creates an identity, user, authentication records for external users
+class ProviderServices
```

Comment:
What do you think about moving the logic from this service class into the `Identity` model? There's not a lot of stuff happening here, just creating two records. It seems like it could just be a method on the identity model.